charts/istio-endpoints/templates/sidecar.yaml

```diff
@@ -6,6 +6,10 @@ metadata:
   name: {{ include "istio-endpoints.fullname" . }}-proxy-config
   labels:
     {{- include "istio-endpoints.labels" . | nindent 4 }}
+  {{- with .Values.sidecar.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
```

Comment:
if this is too much flexibility, we could add more narrowly scoped values (instead of just `annotations`) that would only add the specific annotations for the sync phases / waves?